### ISSUE TITLE
FxCop CA2235 is unaware of primitive types, such as strings, ints or enums

### ISSUE THREAD

**Issue by RalfKoban** [dotnet/roslyn#4592]
Hi,

I'm using Microsoft.CodeAnalysis.FxCopAnalyzers v1.0.1.

I have a field (auto property) which is of type string (or an int, or an enum...).
However, I get a false positive violation of CA2235:

        public string MyProperty { get; }

Best regards,
Ralf

**Comment by srivatsn** [comment]
This has been fixed by https://github.com/dotnet/roslyn-analyzers/pull/261

**Comment by RalfKoban** [comment]
Hi,
I've just updated to "Microsoft.CodeAnalysis.FxCopAnalyzers" v1.1.0 for .NET 4.6 and still getting the 
same error, e.g. for following property:

        public int MaxLength { get; }

So I'm not sure whether this is fixed at all.

Best regards.
Ralf

**Comment by lgolding** [comment]
Reopening so we can re-triage it.

**Comment by bkoelman** [comment]
I get this warning in v1.1.0 in a .NET 4.5 project on these types: 
* int, int?
* bool, bool?
* string
* IList\<string\> (not sure if this is correct)

Also, I'm getting unexpected report on:
* `private static readonly Regex ValueFormatRegex = new...`
* `private const int someField = 5;`
* `public Version Value { get; set; }`

System.Version is [documented](https://msdn.microsoft.com/en-us/library/system.version(v=vs.110).aspx) as serializable. But in VS, MAS does not show [Serializable] on the type. But, the next code correctly roundtrips:

```csharp
using (var stream = new MemoryStream())
{
    BinaryFormatter b = new BinaryFormatter();
    b.Serialize(stream, source);

    stream.Seek(0, SeekOrigin.Begin);
    var result = (VersionParameter) b.Deserialize(stream);
}
```

**Comment by lgolding** [comment]
@RalfKoban @bkoelman 

This is a duplicate of (or at least, it has the same root cause as) #3898. Once that's fixed, this one will automatically be fixed too. So that's the issue you'll want to track.
